### ISSUE TITLE
Unrecognized selector with custom annotation

### ISSUE THREAD

**Issue by bsiddiqui** [nitaliano/react-native-mapbox-gl#512]
Using mapbox on iOS and have several custom annotations.

When adding an annotation after the map has rendered I get this error:
`[RCTMapboxAnnotation annotationImageSource]: unrecognized selector sent instance ...`

**Comment by Kottidev** [comment]
are you fixed this issues ?

**Comment by Kottidev** [comment]
@bsiddiqui

**Comment by bsiddiqui** [comment]
@kottidev nope I haven't

**Comment by markhaasjes** [comment]
I'm using custom annotations and got this error as well. Why is this error being thrown while it does not require an image as prop. I love to see this bug fixed.

**Comment by donnycrash** [comment]
Hey Im also experiencing this issue with a custom annotation using  "react-native": "0.43.3"

**Comment by donnycrash** [comment]
I found a really dirty hack to get by in the meantime... if you comment out line 80 - 85
``` objective-c
    UILongPressGestureRecognizer *longPress = [[UILongPressGestureRecognizer alloc] initWithTarget:self action:@selector(handleLongPress:)];
    [self addGestureRecognizer:longPress];

    UITapGestureRecognizer *singleTap = [[UITapGestureRecognizer alloc] initWithTarget:self action:@selector(handleSingleTap:)];
    singleTap.delegate = self;
    [_map addGestureRecognizer:singleTap];
```
in RCTMapboxGL.m it seems to work as expect (assuming you dont need longpress and single tap on your custom annotation)

**Comment by KevinHu2014** [comment]
experiencing the same issue

**Comment by nico2che** [comment]
i get the same error because i've a null value in my datas which came from redux

```
{
    coordinates: [40.72052634, null], // !!! Here "unrecognized selector" exception
    type: 'point',
    annotationImage: {
        source: { uri: null }, // !!! Or here "unrecognized selector" exception
        height: 25,
        width: 25
    },
    id: 'marker1'
}
```

you have to make conditions to avoid them

**Comment by helderdb** [comment]
Still getting the error while using master version of mapbox...

**Comment by azghibarta** [comment]
@bsiddiqui +1 for this issue, and what's more strange, once it worked, but after a time stopped to work. Those someone figured out how to solve this?

**Comment by alixaxel** [comment]
We are using custom annotations and we are facing the same problem:

```html
<Annotation id={key} key={key} coordinate={{latitude: value.location[0], longitude: value.location[1]}} style={styles.annotation}>
	<TouchableWithoutFeedback onPress={this._onPressButton}>
		<View style={[styles.marker, { backgroundColor: value.color }]} />
	</TouchableWithoutFeedback>
</Annotation>
```

The strange thing is that we are able to render the first 2 debug annotations that we have in the Component constructor, but then if/once we fetch more annotations from our API and set the state to render them, we get the same exception.

<img width="378" alt="screen shot 2017-07-25 at 17 11 36" src="https://user-images.githubusercontent.com/262782/28666125-647fadce-72c6-11e7-938d-e1b716b74152.png">

We are filtering out any nil/empty-like value just to be sure we don't have the same problem @nico2che faced, but it doesn't seem to help at all. We also generated 10,000 points on the constructor of the component and again, we are able to render them without any problem, so I suspect there must be some specific use case when the state changes that triggers this exception to be thrown.

**Comment by helderdb** [comment]
I fixed this by forking and merging PR #555
